src/multitask_personalization/rom/models.py

```diff
@@ -196,3 +197,82 @@ def _visualize_reachable_points(
 
         while True:
             p.stepSimulation(self._physics_client_id)
+
+
+class SphericalROMModel(ROMModel):
+    """ROM model with spherical reachability."""
+
+    def __init__(
+        self,
+        human_spec: HumanSpec,
+        seed: int = 0,
+        radius: float = 0.5,
+    ) -> None:
+        super().__init__()
+        self._radius = radius
+        # Create human
+        # Uncomment for debugging
+        # from pybullet_helpers.gui import create_gui_connection
+        # self._physics_client_id = create_gui_connection()
+        self._physics_client_id = p.connect(p.DIRECT)
+        self._rng = np.random.default_rng(seed)
+        self._human = create_human_from_spec(
+            human_spec, self._rng, self._physics_client_id
+        )
+        origin, _ = self._human.get_pos_orient(self._human.right_wrist)
+        self._sphere_center = origin
+
+        # Uncomment for debugging.
+        # self._reachable_points = self._sample_spherical_points(n=500)
+        # self._visualize_reachable_points()
+
+    def check_position_reachable(
+        self, position: NDArray, padding: float = 1e-6
+    ) -> bool:
+        distance = float(np.linalg.norm(position - self._sphere_center))
+        return distance < self._radius + padding
+
+    def sample_reachable_position(self, rng: np.random.Generator) -> NDArray:
+        return np.array(sample_spherical(self._sphere_center, self._radius, rng))
+
+    def get_reachable_joints(self) -> NDArray:
```

Comment:
if not all ROMModels will implement these methods, should we remove them from the ROMModel definition itself (and make them private methods for ROMModels that do implement them)?